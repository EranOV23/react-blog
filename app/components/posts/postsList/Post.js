import React from 'react';
import moment from 'moment';

export default class Post extends React.Component{
  constructor(props){
    super(props);
    this.post = props.post;
  }

  render(){
    return(
      <li className="post-item">
        <h2>{this.post.title}</h2>
        <p><small className="glyphicon glyphicon-user"></small>By<span className="author">{this.post.author}</span></p>
        <p><small className="glyphicon glyphicon-time"></small>Posted on {moment(parseInt(this.post.date)).format("DD MMM, YYYY") }</p>
        <p>{this.post.description}</p>
        <div className="post-footer">
          <p><strong>Tags: </strong>{this.post.tags.map((tag, i) => {return <span className="tag" key={i}> {tag} </span>})}</p>
          <button className="btn btn-primary">Read more</button>
        </div>
        <hr/>
      </li>
    )
  }
}
