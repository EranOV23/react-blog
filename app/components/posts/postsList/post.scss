@import "../../../assets/css/colors";

.post-item{
  list-style: none;
  width: 30%;
  margin: 10px;
  position: relative;

  .post-image{
    width: 100%;
    img{
      width: 100%;
      border-radius: 20px;
    }
  }

  .post-content{
    position: absolute;
    top: 0;
    opacity: 0;
    background: #eaeaea;
    border-radius: 20px;
    overflow: scroll;
    height: 100%;
    padding: 10px;
    transition: all 0.5s ease;

    h2{
      color: $linkColor;
    }

    .author{
      color: $linkColor;
    }

    .tag{
      background-color: #777;
      color: $fontColor;
      font-size: 12px;
      padding: 4px;
      margin: 2px;
      border-radius: 2px;
    }

    .post-footer{
      display: flex;
      justify-content: space-between;
    }
  }

  .post-content:hover{
    opacity: 1;
  }
}
