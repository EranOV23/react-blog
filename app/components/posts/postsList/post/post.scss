@import "../../../../assets/css/colors";

.post-item{
  list-style: none;
  width: 33%;
  padding: 10px;
  border-radius: border-box;
  position: relative;

  .post-image{
    width: 100%;
    img{
      width: 100%;
      border-radius: 20px;
    }
  }

  .post-content{
    position: absolute;
    top: 0;
    opacity: 0;
    background: white;
    border-radius: 20px;
    overflow-y: scroll;
    overflow-x: hidden;
    height: 100%;
    padding: 10px;
    transition: all 0.5s ease;

    h2{
      color: $linkColor;
    }

    .author{
      color: $linkColor;
    }

    .tag{
      background-color: #777;
      color: $fontColor;
      font-size: 12px;
      padding: 4px;
      margin: 2px;
      border-radius: 2px;
    }

    .post-footer{
      display: flex;
      justify-content: space-between;
    }
  }

  .post-content:hover{
    opacity: 1;
    width: 100%;
  }
}
