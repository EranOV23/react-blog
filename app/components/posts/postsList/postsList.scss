.posts-list{
  padding-left: 0px;
  display: flex;
  flex-wrap: wrap;
}
