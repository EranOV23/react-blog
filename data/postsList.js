module.exports = {
    "posts": [
      {
        "title": "Beach Bum",
        "img": "https://i.pinimg.com/564x/c4/cb/80/c4cb802ab0f22028489d248f61942103.jpg",
        "author": "Eran Ovadia",
        "date": "1421186400000",
        "tags": ["Vodka", "Liqueurs"],
        "mdPath": "data/posts/md/AngularJS - Controllers.md",
        "htmlPath": "data/posts/html/AngularJS - Controllers.html",
        "description": "Meat: 2 oz Grey goose la poire flavored vodka \n Produce: 1 Mint, Fresh \n Condiments: 1 dash Grenadine, 1 oz Lime juice, Fresh \n Beer, Wine & Liquor: 1/2 oz Cointreau, 1/2 oz Maraschino liqueur"
      },
      {
        "title": "Gin Revival",
        "img": "https://i.pinimg.com/564x/6c/cc/4e/6ccc4e1c86b43f1b9e84fcef5f92e7a4.jpg",
        "author": "Eran Ovadia",
        "date": "1421186400000",
        "tags": ["Gin", "Juices"],
        "mdPath":  "data/posts/md/AngularJS - Modules.md",
        "htmlPath": "data/posts/html/AngularJS - Modules.html",
        "description": "The gin revival is big and it's here to stay. We made this delicious pomegranate gin cocktail with Wilderer Fynbos Gin."
      },
      {
        "title": "Ginapple",
        "img": "https://i.pinimg.com/564x/70/0f/46/700f46ea06c717b8e62949390e734b76.jpg",
        "author": "Daniel Kalderon",
        "date": "1421186400000",
        "tags": ["Gin", "Juices"],
        "mdPath": "data/posts/md/AngularJS - Routes.md",
        "htmlPath": "data/posts/html/AngularJS - Routes.html",
        "description": "Apple & Gin Autumn Cocktail | Tart, slightly sweet and smooth with a fruity apple forward taste, this Apple & Gin Cocktail is filled to the brim with warm fall flavors."
      },
      {
        "title": "Melafefonito",
        "img": "https://i.pinimg.com/564x/91/91/d2/9191d251e7053f245e17e3281e65bc01.jpg",
        "author": "Daniel Kalderon",
        "date": "1421186400000",
        "tags": ["Gin", "Juices"],
        "mdPath": "data/posts/md/AngularJS - Services.md",
        "htmlPath": "data/posts/html/AngularJS - Services.html",
        "description": "This twist on the classic gin and cucumber cocktail is sure to refresh you on any day of the week! Light, delicious and pretty easy to make!"
      },
      {
        "title": "Tequila Sunrise",
        "img": "https://i.pinimg.com/564x/6e/cc/3b/6ecc3be27ce4c07808f1a542b8756ae1.jpg",
        "author": "Daniel Kalderon",
        "date": "1418162400000",
        "tags": ["Tequila"],
        "mdPath": "data/posts/md/Grunt - Custom Tasks.md",
        "htmlPath": "data/posts/html/Grunt - Custom Tasks.html",
        "description": "Condiments: 1/2 oz Grenadine \n Drinks: 4 oz Orange juice \n Frozen: 1 Ice cubes \n Desserts: 1 Maraschino cherries \n Beer, Wine & Liquor: 2 oz Tequila \n Other: orange slices, for garnishing umbrellas, optional for garnishing"
      },
      {
        "title": "Pisco sour",
        "img": "https://i.pinimg.com/564x/f3/16/c8/f316c8f6505f1c9935df2c6b94164821.jpg",
        "author": "Eran Ovadia",
        "date": "1418162400000",
        "tags": ["Pisco", "Liqueurs"],
        "mdPath": "data/posts/md/Grunt - Intro.md",
        "htmlPath": "data/posts/html/Grunt - Intro.html",
        "description": "Produce: 1 Lime, slices \n Refrigerated: 1 Egg white \n Condiments: 50 ml Lime juice 50 ml Sugar syrup \n Frozen: 1 large handful Ice \n Beer, Wine & Liquor: 3 drops Angostura bitters \n Other: 50ml Pisco"
      },
      {
        "title": "Arak Hanamal 40",
        "img": "https://i.pinimg.com/564x/89/91/f7/8991f71b87d94ab822bad8657cd8a4ae.jpg",
        "author": "Eran Ovadia",
        "date": "1447557600000",
        "tags": ["Anise"],
        "mdPath": "data/posts/md/jQuery - Events, AJAX.md",
        "htmlPath": "data/posts/html/jQuery - Events, AJAX.html",
        "description": " One bottle HaNamal 40 Extra fine arak 40% 700ml Produced."
      },
      {
        "title": "Gin Hendrix",
        "img": "https://i.pinimg.com/564x/fa/a9/90/faa99081c684143f6649ebccdb9f0485.jpg",
        "author": "Eran Ovadia",
        "date": "1438394800000",
        "tags": ["Gin"],
        "mdPath": "data/posts/md/jQuery - Selectors, DOM.md",
        "htmlPath": "data/posts/html/jQuery - Selectors, DOM.html",
        "description": "Hendrick's Gin is a brand of gin produced by William Grant & Sons in Girvan, Scotland, and launched in 1999."
      }
    ]
  };
