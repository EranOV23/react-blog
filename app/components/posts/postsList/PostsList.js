import React from 'react';
import {connect} from 'react-redux';
import Post from './Post';

class PostsList extends React.Component {
    constructor(props){
      super(props);
    }


    renderPosts(post , i){
      return <Post post={post} key={i}/>
    }

    render() {
        return (
          <ul className="posts-list">
            {this.props.posts.slice(this.props.sliceStart, this.props.sliceEnd).map( (post, i) => this.renderPosts(post, i) )}
          </ul>
        )
    }
}


function mapStateToProps(state) {
    return {
      posts: state.posts,
    }
}

export default connect(mapStateToProps, null)(PostsList);
