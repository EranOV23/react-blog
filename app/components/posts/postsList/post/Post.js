import React from 'react';
import moment from 'moment';
import { NavLink } from 'react-router-dom';
import './post.scss';


export default class Post extends React.Component{
  constructor(props){
    super(props);
  }

  render(){
    return(
      <li className="post-item">
        <div className="post-image">
          <img src={this.props.post.img} alt="post-image"/>
        </div>

        <div className="post-content">
          <NavLink to={`/post/${this.props.post.title}`}>
            <h2>{this.props.post.title}</h2>
          </NavLink>
          <p><small className="glyphicon glyphicon-user"></small> By <span className="author">{this.props.post.author}</span></p>
          <p><small className="glyphicon glyphicon-time"></small> Posted on {moment(parseInt(this.props.post.date)).format("DD MMM, YYYY") }</p>
          <p>{this.props.post.description}</p>
          <div className="post-footer">
            <p><strong>Tags: </strong>{this.props.post.tags.map((tag, i) => {return <span className="tag" key={i}> {tag} </span>})}</p>
          </div>
          <NavLink to={`/post/${this.props.post.title}`}>
            <button className="btn btn-primary">Read more</button>
          </NavLink>
        </div>
      </li>
    )
  }
}
