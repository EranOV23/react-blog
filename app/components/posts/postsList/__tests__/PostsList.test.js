import React from 'react';
import PostsList from '../PostsList';
import '../postsList.scss';
import '../post.scss';

// import renderer from 'react-test-renderer';
import { shallow, configure  } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('PostsList', () => {

  const testProps = {
    posts: [
      {
        title: "AngularJS - Modules",
        author: "Ilan Cohen",
        date: "1421186400000",
        tags: ["JavaScript", "AngularJS"],
        mdPath:  "data/posts/md/AngularJS - Modules.md",
        htmlPath: "data/posts/html/AngularJS - Modules.html",
        description: "You can think of a module as a container"
      }
    ],
    query: "",
  };

  it('renders correctly' , () => {
    const wrapper = shallow(
      <PostsList {...testProps} />
    );


    expect(wrapper.find('Post').length).toBe(1);

    expect(wrapper).toMatchSnapshot();
  });

});