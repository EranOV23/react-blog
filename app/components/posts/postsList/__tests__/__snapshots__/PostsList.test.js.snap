// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostsList
    posts={
      Array [
        Object {
          "author": "Ilan Cohen",
          "date": "1421186400000",
          "description": "You can think of a module as a container",
          "htmlPath": "data/posts/html/AngularJS - Modules.html",
          "mdPath": "data/posts/md/AngularJS - Modules.md",
          "tags": Array [
            "JavaScript",
            "AngularJS",
          ],
          "title": "AngularJS - Modules",
        },
      ]
    }
    query=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Post
          post={
            Object {
              "author": "Ilan Cohen",
              "date": "1421186400000",
              "description": "You can think of a module as a container",
              "htmlPath": "data/posts/html/AngularJS - Modules.html",
              "mdPath": "data/posts/md/AngularJS - Modules.md",
              "tags": Array [
                "JavaScript",
                "AngularJS",
              ],
              "title": "AngularJS - Modules",
            }
          }
        />,
      ],
      "className": "posts-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "post": Object {
            "author": "Ilan Cohen",
            "date": "1421186400000",
            "description": "You can think of a module as a container",
            "htmlPath": "data/posts/html/AngularJS - Modules.html",
            "mdPath": "data/posts/md/AngularJS - Modules.md",
            "tags": Array [
              "JavaScript",
              "AngularJS",
            ],
            "title": "AngularJS - Modules",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Post
            post={
              Object {
                "author": "Ilan Cohen",
                "date": "1421186400000",
                "description": "You can think of a module as a container",
                "htmlPath": "data/posts/html/AngularJS - Modules.html",
                "mdPath": "data/posts/md/AngularJS - Modules.md",
                "tags": Array [
                  "JavaScript",
                  "AngularJS",
                ],
                "title": "AngularJS - Modules",
              }
            }
          />,
        ],
        "className": "posts-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "post": Object {
              "author": "Ilan Cohen",
              "date": "1421186400000",
              "description": "You can think of a module as a container",
              "htmlPath": "data/posts/html/AngularJS - Modules.html",
              "mdPath": "data/posts/md/AngularJS - Modules.md",
              "tags": Array [
                "JavaScript",
                "AngularJS",
              ],
              "title": "AngularJS - Modules",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
